### ISSUE TITLE
Transpose and Hermitian adjoint are switched for LinearOperator

### ISSUE THREAD

**Issue by RendersJens** [scipy/scipy#13734]
For some reason, the definition of Hermition adjoint and Transpose ar switched in the implementation of LinearOperator

#### Reproducing code example:

This simple operator shows the problem
```
class TestOperator(LinearOperator):
    def __init__(self, n):
        self.shape = (n, n)
        self.dtype = np.complex128

    def _matvec(self, x):
        return 1j*x

    def _rmatvec(self, x):
        return 1j*x

A = TestOperator(3)
```
Now A is equivalent to the complex matrix
```
[i, 0, 0]
[0, i, 0]
[0, 0, i]
```
Therefore A.T should be equal to A, and A.H should be its conjugate: -A

Instead we get the opposite:
```
>>> A.T @ np.eye(3)
array([[0.-1.j, 0.-0.j, 0.-0.j],
       [0.-0.j, 0.-1.j, 0.-0.j],
       [0.-0.j, 0.-0.j, 0.-1.j]])

>>> A.H @ np.eye(3)
array([[0.+1.j, 0.+0.j, 0.+0.j],
       [0.+0.j, 0.+1.j, 0.+0.j],
       [0.+0.j, 0.+0.j, 0.+1.j]])

```

This should be easy to fix. The issue is [here](https://github.com/scipy/scipy/blob/master/scipy/sparse/linalg/interface.py). The definition of `_AdjointLinearOperator` (line 553) and ` _TransposedLinearOperator` (line 573) should be switched
#### Scipy/Numpy/Python version information:

Current master branch.

**Comment by RendersJens** [comment]
This behaviour is actually correct according to the documentation. It stated there that `_rmatvec` should implement `A^H @ x`, multiplication with the hermitian adjoint.

This still seems troublesome to me. The name `_rmatvec` seems to imply that it is right matrix vector multiplication, which is `x^T @ A = A^T @ x`, no conjugate there. It makes little sense that the numpy function `numpy.conj` is invoked when only working  with real operators and the regular transpose. This is the case now.

**Comment by nschloe** [comment]
I agree. `rmatvec` leans too much on `__rmatmul__` which is doing a different thing. It should probably have been called `adjmatvec`.
